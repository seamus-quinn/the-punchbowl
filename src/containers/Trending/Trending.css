.trending {
  display: flex;
  flex-wrap: wrap;
  margin: 2% 3% 190px 0;
  justify-content: center;
}

.button {
  color: #fff;
  background: none;
  font-family: 'Raleway', sans-serif;
  font-size: 24px;
  border: none;
  width: 200px;
  height: 60px;
  margin: 4px;
}