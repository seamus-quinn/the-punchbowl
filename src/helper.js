export const makeWordsArr = (arr) => {
  const words = arr.reduce((acc, article) => {
    if (article.description) {
      const title = article.title.split(' ')
      const description = article.description.split(' ')
      acc.push(...title, ...description)
    } else {
      const title = article.title.split(' ')
      acc.push(...title)
    }
    return acc;
  }, [])
  return words;
}

export const countWords = (arr) => {
  const countObj = arr.reduce((acc, word) => {
    const lowerWord = word.toLowerCase();
    if (!acc[lowerWord]) {
      acc[lowerWord] = 0
    }
    acc[lowerWord]++
    return acc;
  }, {})
  return countObj;
}

export const cleanWords = (arr) => {
  const commonWords = ['the', 'of', 'to', 'in', 'a', 'and', 'for', 'on', 'with', 'new', 'at', 'is', 'his', 'man', 'as', '-', '_', 'â€”', 'you', 'that', 'into', 'about', 'over', 'an', 'are', 'after', 'be', 'from', 'it', 'one', 'her', 'who', 'the', 'of', 'and', 'a', 'to', 'in', 'is', 'you', 'that', 'it', 'he', 'was', 'for', 'on', 'are', 'as', 'with', 'his', 'they', 'I', 'at', 'be', 'this', 'have', 'from', 'or', 'one', 'had', 'by', 'word', 'but', 'not', 'what', 'all', 'were', 'we', 'when', 'your', 'can', 'said', 'there', 'use', 'an', 'each', 'which', 'she', 'do', 'how', 'their', 'if', 'will', 'up', 'other', 'about', 'out', 'many', 'then', 'them', 'these', 'so', 'some', 'her', 'would', 'make', 'like', 'him', 'into', 'time', 'has', 'look', 'two', 'more', 'write', 'go', 'see', 'number', 'no', 'way', 'could', 'people', 'my', 'than', 'first', 'been', 'call', 'who', 'its', 'now', 'find', 'long', 'down', 'day', 'did', 'get', 'come', 'made', 'may', 'part', '--', '(ap)', 'says', 'news', 'went', 'times', 'fox']
  commonWords.forEach(commonWord => {
    arr.forEach(word => {
      if (word === commonWord) {
        arr.splice(arr.indexOf(word), 1)
      }
    })
  })
  return arr;
}

export const cleanSource = (source) => {
  if(source === 'Npr.org') {
    return 'NPR'
  } else if (source === 'Theatlantic.com') {
    return 'The Atlantic'
  } else if (source === 'Ap.org') {
    return 'The Associated Press'
  } else if (source === 'C-span.org') {
    return 'C-SPAN'
  } else if (source === 'Cato.org') {
    return 'The Cato Institute'
  } else {
    return source
  }
}

export const createMatchObject = (keywords, articles) => ({
  keywords,
  articles,
  id: Date.now()
})

export const flattenArrays = (arr) => {
  return arr.reduce((acc, arr) => {
    return acc.concat(Array.isArray(arr) ? flattenArrays(arr) : arr);
  }, []);
}

export const cleanArticles = (articles) => {
  const cleanedArticles = articles.map(article => {
    let cleanedArticle = {}
    if (!article.source) {
      cleanedArticle.source = {name: 'Unable to find article source'}
    } else {
      cleanedArticle.source = article.source
    }
    if (!article.title) {
      cleanedArticle.title = 'Unable to find article title'
    } else {
      cleanedArticle.title = article.title
    }
    if (!article.description) {
      cleanedArticle.description = 'Unable to find article description'
    } else {
      cleanedArticle.description = article.description
    }
    if (!article.url) {
      cleanedArticle.url = 'Unable to find link to article'
    } else {
      cleanedArticle.url = article.url
    }
    if (!article.urlToImage) {
      cleanedArticle.urlToImage = null
    }
    else {
      cleanedArticle.urlToImage = article.urlToImage
    }
    return cleanedArticle
  })
  return cleanedArticles
} 

export const cleanImageUrl = (url) => {
  if(!url) {
    return false;
  }
  const split = url.split('/')
  if(split[0] === 'https:' || split[0] === 'http:'){
    return true
  } else {
    return false
  }
}

export const topTen = (arr) => {
  const words = helper.makeWordsArr(arr)
  const wordObj = helper.countWords(words);
  const keys = Object.keys(wordObj);
  const sorted = keys.sort((a, b) => {
    return wordObj[b] - wordObj[a];
  })
  const cleaned = helper.cleanWords(sorted)
  return cleaned.slice(0, 10)
}