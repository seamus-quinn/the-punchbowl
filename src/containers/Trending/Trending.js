import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as helper from '../../helper';
import './Trending.css'

class Trending extends Component {
  constructor(props) {
    super(props)

    this.state = {
      trendingWords: [],
      loading: ''
    }
  }

  componentDidMount() {
    if (!this.props.articles.length) {
      this.setState({ loading: 'wow such load' })
    }
    const trendingWords = helper.topTen(this.props.articles)
    this.setState({ trendingWords })
  }

  render() {
    const { trendingWords, loading } = this.state;
    const trending = trendingWords.map((word, index) => {
      return(
        <button 
          className='button'
          onClick={() => this.props.populateInputField(word)}
          key={index}
        >+ {word}</button>
      ) 
    })
    return (
      <div className='trending'>
        {loading !== '' ? loading : trending}
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  articles: state.articles
})

Trending.propTypes = {
  articles: PropTypes.array
}

export {
  Trending,
  mapStateToProps
}

export default connect(mapStateToProps)(Trending)