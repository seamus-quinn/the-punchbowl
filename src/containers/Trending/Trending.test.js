import React from 'react';
import { Trending } from './Trending'
import { shallow } from 'enzyme';
import * as helper from '../../helper'

describe('Trending', () => {

  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      keywords: ['wow', 'such', 'garbage'],
      articles: [
        {
          title: 'Bummer Dude',
          description: 'I forgot my charger at Turing and now I have to go back :('
        }
      ]
    }
    wrapper = shallow(<Trending {...mockProps} />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('componentDidMount', () => {

    it('sets the state to loading if the props inherited are not preseent', () => {

    })

    it('calls topTen with the correct arguments', () => {
      
    })

    it('sets the state of trendingWords once it receives props', () => {

    })
  })


})