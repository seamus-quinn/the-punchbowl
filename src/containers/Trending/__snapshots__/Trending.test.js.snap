// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trending matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Trending
    articles={
        Array [
            Object {
              "description": "I forgot my charger at Turing and now I have to go back :(",
              "title": "Bummer Dude",
            },
          ]
    }
    keywords={
        Array [
            "wow",
            "such",
            "garbage",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="button"
          onClick={[Function]}
>
          + 
          i
</button>,
        <button
          className="button"
          onClick={[Function]}
>
          + 
          turing
</button>,
        <button
          className="button"
          onClick={[Function]}
>
          + 
          bummer
</button>,
        <button
          className="button"
          onClick={[Function]}
>
          + 
          forgot
</button>,
        <button
          className="button"
          onClick={[Function]}
>
          + 
          charger
</button>,
        <button
          className="button"
          onClick={[Function]}
>
          + 
          dude
</button>,
        <button
          className="button"
          onClick={[Function]}
>
          + 
          back
</button>,
        <button
          className="button"
          onClick={[Function]}
>
          + 
          :(
</button>,
      ],
      "className": "trending",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "+ ",
            "i",
          ],
          "className": "button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "+ ",
          "i",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "+ ",
            "turing",
          ],
          "className": "button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "+ ",
          "turing",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "+ ",
            "bummer",
          ],
          "className": "button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "+ ",
          "bummer",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "+ ",
            "forgot",
          ],
          "className": "button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "+ ",
          "forgot",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "+ ",
            "charger",
          ],
          "className": "button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "+ ",
          "charger",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "+ ",
            "dude",
          ],
          "className": "button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "+ ",
          "dude",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "+ ",
            "back",
          ],
          "className": "button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "+ ",
          "back",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "+ ",
            ":(",
          ],
          "className": "button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "+ ",
          ":(",
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="button"
            onClick={[Function]}
>
            + 
            i
</button>,
          <button
            className="button"
            onClick={[Function]}
>
            + 
            turing
</button>,
          <button
            className="button"
            onClick={[Function]}
>
            + 
            bummer
</button>,
          <button
            className="button"
            onClick={[Function]}
>
            + 
            forgot
</button>,
          <button
            className="button"
            onClick={[Function]}
>
            + 
            charger
</button>,
          <button
            className="button"
            onClick={[Function]}
>
            + 
            dude
</button>,
          <button
            className="button"
            onClick={[Function]}
>
            + 
            back
</button>,
          <button
            className="button"
            onClick={[Function]}
>
            + 
            :(
</button>,
        ],
        "className": "trending",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "+ ",
              "i",
            ],
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "+ ",
            "i",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "+ ",
              "turing",
            ],
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "+ ",
            "turing",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "+ ",
              "bummer",
            ],
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "+ ",
            "bummer",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "+ ",
              "forgot",
            ],
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "+ ",
            "forgot",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "+ ",
              "charger",
            ],
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "+ ",
            "charger",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "+ ",
              "dude",
            ],
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "+ ",
            "dude",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "+ ",
              "back",
            ],
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "+ ",
            "back",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "+ ",
              ":(",
            ],
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "+ ",
            ":(",
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
